app/license.go

```diff
@@ -379,7 +379,10 @@ func (s *Server) GenerateRenewalToken(expiration time.Duration) (string, *model.
 
 	currentToken, _ := s.Store.System().GetByName(model.SystemLicenseRenewalToken)
 	if currentToken != nil {
-		tokenIsValid, _ := s.renewalTokenValid(currentToken.Value, license.Customer.Email)
+		tokenIsValid, err := s.renewalTokenValid(currentToken.Value, license.Customer.Email)
+		if err != nil {
+			mlog.Warn("error checking license renewal token validation", mlog.Err(err))
+		}
```

Comment:
Looking more into the errors reported in the load testing `token is expired by`, this is actually returned from the [JWT library here](https://github.com/mattermost/mattermost-server/blob/baf9f69c273c67d8c1569c6bab9376d21971d8cf/vendor/github.com/dgrijalva/jwt-go/claims.go#L40). Which means that we are reporting an error for the token being expired which we know it will eventually happen given that it has a time period that it is valid and we are saving it for some reason. 

Again, I don't think there is any value on logging a warning for something that is expected and we should just be generating a new token each time - like I mentioned above. 

I'm 4/5 that we should **not** keep this error as is. We should either:

1. Stop storing the token in the System Store and just generate the token on demand. This would avoid storing something that will expire and produce these unnecessary errors/checks. 
2. Surface the error type [ValidationErrorExpired](https://github.com/mattermost/mattermost-server/blob/baf9f69c273c67d8c1569c6bab9376d21971d8cf/vendor/github.com/dgrijalva/jwt-go/claims.go#L41) from our `renewalTokenValid` and check here so we can ignore

Reply:
Fantastic analysis Maria! Yeah, fully agree we can ignore the expired errors.

Reply:
@marianunez went with option 1